### DISCUSSION TITLE
Duplicate vs Copy & Paste

### DISCUSSION THREAD

**Discussion by SJK55** [moneymanagerex/moneymanagerex#5383] [Ideas]
I used Duplicate on a transaction. Depending what settings you have set in [Options] - [Others] - [Default Date] is whether you get the date of the transaction duplicated or the current date. - Fine.
There is a another option of Copy Transaction and Paste Transaction.  In my mind copy and paste means just that, it shouldn't

 change the date of a copied transaction. This feature of copy and paste is no different to the duplicate transaction. 
So why have both (3) options in the right mouse click on a transaction?
Is it where a true copy and paste, the date would keep the date being copied. (This is my perception of how it should work)

Any thoughts?

![Uploading Copy - Paste vs Duplicate MoneyManager EX.jpg…]()

**Comment by SJK55** [comment]
![Copy - Paste vs Duplicate MoneyManager EX](https://user-images.githubusercontent.com/14268231/204119061-816876ee-cb0d-4385-aeb5-cea937f8b54c.jpg)

**Reply by SJK55** [reply]
Had trouble adding image. So added here.

**Comment by tactilis** [comment]
@SJK55
> So why have both (3) options in the right mouse click on a transaction?

Duplicate and Copy/Paste are completely different operations.

**Duplicate**
1. Works for a single transaction only (the menu item is greyed out if you have more than one row selected).
2. Opens the Transaction dialog with the fields pre-populated with the data from the transaction you are duplicating (but note the comments on Date below).
3. You are able to modify the fields if you wish.
4. Clicking OK saves the new (duplicated) transaction. Clicking Cancel just cancels the dialog, so no new transaction is created.

**Copy/Paste**
1. Copy Transaction(s): Copies to the clipboard the currently selected single or multiple transaction(s).
2. The standard shortcut key CTRL+C can be used to copy transactions.
3. Paste Transaction(s): Pastes the transaction(s) from the clipboard, creating new ones with the same data (but note the comments on Date below) in the account where you do the paste.
4. The standard shortcut key CTRL+V can be used to paste transactions.
5. If you paste the transaction(s) into a different MMEX account to that from which they were copied, specific logic is applied to adjust the Account field (or the From & To account fields in the case of Transfers) to cater for the fact that they now reside in a different account.
6. Because the transaction(s) are copied to the clipboard, they can be pasted into other applications (Excel, Word, Notepad, etc) as tab separated fields.

**Dates**
The Options shown below determine whether the original Date is used for Duplicate and Paste operations.
 
![image](https://user-images.githubusercontent.com/4060092/204127791-7666b7c4-4988-466f-8d1a-601f7ac07dcb.png)

**Reply by SJK55** [reply]
Thanks Martin, though I've used the application a lot, there are always hidden or forgotten options I miss.
Thanks for explaining that very succinctly and I have a better understanding of why the copy and paste exist now.
